fiftyone/core/dataset.py

```diff
@@ -2617,7 +2618,11 @@ def _upsert_samples_batch(
                 d.pop("_id", None)
                 ops.append(InsertOne(d))  # adds `_id` to dict
 
-        foo.bulk_write(ops, self._sample_collection, ordered=False)
+        try:
+            self._sample_collection.bulk_write(ops, ordered=False)
+        except BulkWriteError as bwe:
+            msg = bwe.details["writeErrors"][0]["errmsg"]
+            raise ValueError(msg) from bwe
 
```

Comment:
good call, I did this also in my local impl so that we can avoid the further batching in `foo.bulk_write()` and not overcomplicate that function.